pf/tfbridge/provider_plan.go

```diff
@@ -68,5 +70,12 @@ func (p *provider) plan(
 		return nil, err
 	}
 
-	return planResp, nil
+	var diags multierror.Error
```

Comment:
Won't block on it but this looks useful.

Does it need to be special cased or it's just a way to handle things? there is this typical stanza I try to convert everything into:

```
if err := p.processDiagnostics(planResp.Diagnostics); err != nil {
   return err
}
```

If that works perhaps your handlers could go into improving processDiagnostics implementation? Multierror sounds useful.

We also need to hook up logging but it's a different ticket.